packages/emitter-framework/src/python/components/enum-declaration/enum-declaration.test.tsx

```diff
@@ -0,0 +1,302 @@
+import { efRefkey } from "#python/utils/refkey.js";
+import { Tester } from "#test/test-host.js";
+import { d } from "@alloy-js/core/testing";
+import * as py from "@alloy-js/python";
+import { t } from "@typespec/compiler/testing";
+import { describe, expect, it } from "vitest";
+import { getOutput } from "../../test-utils.js";
+import { EnumDeclaration } from "./enum-declaration.js";
+
+describe("Python Enum Declaration", () => {
+  it("takes an enum type parameter", async () => {
+    const { program, Foo } = await Tester.compile(t.code`
+      enum ${t.enum("Foo")} {
+        one: 1,
+        two: 2,
+        three: 3
+      }
+    `);
+
+    expect(getOutput(program, [<EnumDeclaration type={Foo} />])).toRenderTo(d`
+      from enum import IntEnum
+
+      class Foo(IntEnum):
+        ONE = 1
+        TWO = 2
+        THREE = 3
+
+
+    `);
+  });
+
+  it("adds Python doc from TypeSpec", async () => {
+    const { program, Foo } = await Tester.compile(t.code`
+      /**
+       * This is a test enum
+       */
+      enum ${t.enum("Foo")} {
+        @doc("This is one")
+        one: 1,
+        two: 2,
+        three: 3
+      }
+    `);
+    const output = getOutput(program, [<EnumDeclaration type={Foo} />]);
+
+    expect(output).toRenderTo(d`
+      from enum import IntEnum
+
+      class Foo(IntEnum):
+        """
+        This is a test enum
+        """
+
+        ONE = 1  #: This is one
```

Comment:
I think formatting the docs like this might hit line-length issues pretty quickly - I would recommend using this:
```python
from enum import IntEnum
      class Foo(IntEnum):
        """This is a test enum"""
        ONE = 1
        """This is one"""
        TWO = 2
        """This is just two, but we'll
        give it a much longer description.
        """
```